src/main/app/case/from-api-format.ts

```diff
@@ -5,16 +5,16 @@ import { Case, Checkbox, YesOrNo, formFieldsToCaseMapping, formatCase } from './
 
 const fields = {
   ...invert(formFieldsToCaseMapping),
-  D8MarriageIsSameSexCouple: data => ({
+  D8MarriageIsSameSexCouple: (data: Partial<CaseData>): Partial<Case> => ({
     sameSex: data.D8MarriageIsSameSexCouple === YesOrNo.Yes ? Checkbox.Checked : Checkbox.Unchecked,
   }),
-  D8MarriageDate: data => ({
+  D8MarriageDate: (data: Partial<CaseData>): Partial<Case> => ({
     relationshipDate: fromApiDate(data.D8MarriageDate),
   }),
 };
 
-const fromApiDate = (date: string) => {
-  const [year, month, day] = date.split('-');
+const fromApiDate = (date = '') => {
+  const [year = '', month = '', day = ''] = date.split('-');
```

Comment:
is this a good idea? maybe returning undefined would be better? I dunno

Reply:
👍 Good idea, I've refactored this in 0f6aed0